dashboard-app/src/lib/ui/components/Boundaries/ErrorBoundary.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { AxiosError } from 'axios';
+import { propOr } from 'ramda';
+
+
+type ErrorBoundaryState = {
+  hasError: boolean
+  isHttpError: boolean
+  statusCode: number
+  message: string
+};
+
+const isHttpError = (error: Error) =>
+  ['isAxiosError', 'response']
+    .reduce((acc, s) => acc || error.hasOwnProperty(s), false);
+
+class Boundary<T extends {}> extends React.Component<T, ErrorBoundaryState> {
+  constructor (props: Readonly<T>) {
+    super(props);
+    this.state = {
+      hasError: false,
+      isHttpError: false,
+      statusCode: 0,
+      message: '',
+    };
+  }
+
+  static getDerivedStateFromError (error: Error | AxiosError) {
+    return {
+      hasError: true,
+      isHttpError: isHttpError(error),
+      statusCode: propOr(0, 'code', error),
+      message: error.message,
+    };
+  }
+
+  componentDidCatch (error: Error, errorInfo: { componentStack: any }) {
+    if (process.env.NODE_ENV !== 'test') {
+      console.log(error, errorInfo);
+    }
+  }
+
+  renderError () {
+    return (
+      <div>
```

Comment:
NOTE!

I haven't styled anything here because I don't know what we want this to look like.

Reply:
The text here isn't loading for me locally. I get a blank screen when running production build with the following console errors:
```
Error: ""
    er index.tsx:91
    React 8
        Zi
        Ro
        Ga
        Xa
        js
        Cs
        Ns
        Mr
react-dom.production.min.js:4408
    React 7
        ha
        callback
        oa
        ia
        Wa
        qa
        Ds
    unstable_runWithPriority scheduler.production.min.js:274
    React 5
        Ds
        js
        Cs
        Ns
        Mr
```

Reply:
I get this:

<img width="1280" alt="Screenshot 2019-07-12 at 15 23 18" src="https://user-images.githubusercontent.com/12719553/61135505-aa9f9180-a4b9-11e9-8b59-4e4cfeab6d7a.png">

Reply:
How are you creating an error?

Reply:
Either by setting `error = new Error('oops')` in the component or by throwing from some component in the child component tree.